Mathlib/RingTheory/Ideal/Pointwise.lean

```diff
@@ -0,0 +1,115 @@
+/-
+Copyright (c) 2024 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import Mathlib.RingTheory.Ideal.Maps
+
+
+/-! # Pointwise instances on `Ideal`s
+
+This file provides the action `Ideal.pointwiseMulAction` which morally matches the action of
+`mulActionSet` (though here an extra `Ideal.span` is inserted).
+
+This actions is available in the `Pointwise` locale.
+
+## Implementation notes
+
+This file is similar (but not identical) to `RingTheory/Subring/Pointwise.lean`.
+Where possible, try to keep them in sync.
+
+-/
+
+
+open Set
+
+variable {M R : Type*}
+
+namespace Ideal
+
+section Monoid
+
+variable [Monoid M] [CommRing R] [MulSemiringAction M R]
+
+/-- The action on an ideal corresponding to applying the action to every element.
+
+This is available as an instance in the `Pointwise` locale. -/
+protected def pointwiseMulSemiringAction : MulSemiringAction M (Ideal R) where
+  smul a I := Ideal.map (MulSemiringAction.toRingHom _ _ a) I
+  one_smul I :=
+    congr_arg (I.map ·) (RingHom.ext <| one_smul M) |>.trans I.map_id
+  mul_smul _a₁ _a₂ I :=
+    congr_arg (I.map ·) (RingHom.ext <| mul_smul _ _) |>.trans (I.map_map _ _).symm
+  smul_one a := by simp only [Ideal.one_eq_top]; exact Ideal.map_top _
+  smul_mul a I J := Ideal.map_mul (MulSemiringAction.toRingHom _ _ a) I J
+  smul_add a I J := Ideal.map_sup _ I J
+  smul_zero a := Ideal.map_bot
+
+scoped[Pointwise] attribute [instance] Ideal.pointwiseMulSemiringAction
+
+open Pointwise
+
+theorem pointwise_smul_def {a : M} (S : Ideal R) :
+    a • S = S.map (MulSemiringAction.toRingHom _ _ a) :=
+  rfl
+
+-- note: unlike with `Subring`, `pointwise_smul_toAddSubgroup` wouldn't be true
```

Comment:
It's true for group actions though, right? Should it be in the `Group` part of the file? Similar comment for `coe_pointwise_smul`.

This observation is perhaps an indication that the correct definition for monoids is to use comap and let them act on the right? But I'm not advocating for this!

Reply:
I added a weaker version.